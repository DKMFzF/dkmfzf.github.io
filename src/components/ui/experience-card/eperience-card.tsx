import { FC } from 'react';
import styles from './eperience-card.module.scss';

// TODO: вынести общии стили в компонент about-me-article

export const ExperienceCardUI: FC = () => (
	<div className={styles.experienceCard}>
		<div className={styles.experienceCard__experienceContainer}>
			<span className={styles.experienceCard__number}>2</span>
			<span className={styles.experienceCard__yearsLabel}>years</span>
		</div>
		<div className={styles.experienceCard__containerFooter}>
			<p className={styles.experienceCard__description}>
				I am immersed in the IT community, participate in hackathons,
				take courses and study at the SFU Institute
			</p>
			<div className={styles.experienceCard__timelineArrow}>-{'>'}</div>
		</div>
	</div>
);
