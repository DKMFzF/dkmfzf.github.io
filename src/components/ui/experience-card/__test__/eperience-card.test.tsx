import { testRenderComponent } from '@utils-testing';
import { ExperienceCardUI } from '../eperience-card';

describe('[ExperienceCardUI]', () => {
	it('Renders experience-card correctly and matches snapshot', () =>
		testRenderComponent(<ExperienceCardUI />));
});
