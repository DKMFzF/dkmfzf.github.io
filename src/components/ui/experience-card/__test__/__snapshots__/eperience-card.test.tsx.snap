// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ExperienceCardUI] Renders experience-card correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="experience-card"
  >
    <div
      class="experience-card__experience-container"
    >
      <span
        class="experience-card__number"
      >
        2
      </span>
      <span
        class="experience-card__years-label"
      >
        years
      </span>
    </div>
    <div
      class="experience-card__container-footer"
    >
      <p
        class="experience-card__description"
      >
        I am immersed in the IT community, participate in hackathons, take courses and study at the SFU Institute
      </p>
      <div
        class="experience-card__timeline-arrow"
      >
        &gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;
